File: remove_prints.py
#!/usr/bin/env python3
"""
Remove all print() and debugPrint() statements from lib/core folder
Keeps if (kDebugMode) blocks but removes only the prints inside
Preserves app_logger.dart as it's a logging utility
"""

import os
import re
from pathlib import Path

def remove_prints_from_file(file_path):
    """Remove print statements from a single file"""

    # Skip app_logger.dart - it's a logging utility
    if 'app_logger.dart' in str(file_path):
        print(f"⏭️  Skipping {file_path} (logging utility)")
        return 0

    try:
        with open(file_path, 'r', encoding='utf-8') as f:
            content = f.read()

        original_content = content
        removed_count = 0

        # Pattern 1: Remove standalone print() statements
        # Matches: print('...');
        pattern1 = r"(\s*)print\([^)]*\);"
        matches1 = re.findall(pattern1, content)
        removed_count += len(matches1)
        content = re.sub(pattern1, r"\1// Debug logging removed", content)

        # Pattern 2: Remove debugPrint() statements
        pattern2 = r"(\s*)debugPrint\([^)]*\);"
        matches2 = re.findall(pattern2, content)
        removed_count += len(matches2)
        content = re.sub(pattern2, r"\1// Debug logging removed", content)

        # Pattern 3: Remove multi-line print statements
        # Matches: print('...'
        #               '...');
        pattern3 = r"(\s*)print\([^;]*\);"
        content = re.sub(pattern3, r"\1// Debug logging removed", content, flags=re.MULTILINE | re.DOTALL)

        # Pattern 4: Remove multi-line debugPrint statements
        pattern4 = r"(\s*)debugPrint\([^;]*\);"
        content = re.sub(pattern4, r"\1// Debug logging removed", content, flags=re.MULTILINE | re.DOTALL)

        # Only write if content changed
        if content != original_content:
            with open(file_path, 'w', encoding='utf-8') as f:
                f.write(content)
            print(f"✅ {file_path}: Removed ~{removed_count} print statements")
            return removed_count
        else:
            print(f"⏭️  {file_path}: No prints found")
            return 0

    except Exception as e:
        print(f"❌ Error processing {file_path}: {e}")
        return 0

def main():
    # Path to lib/core folder
    core_path = Path("ruwaq_jawi/lib/core")

    if not core_path.exists():
        print(f"❌ Path not found: {core_path}")
        return

    print("🚀 Starting print statement removal from lib/core")
    print("=" * 60)

    total_removed = 0
    files_processed = 0

    # Process all .dart files in lib/core
    for dart_file in core_path.rglob("*.dart"):
        removed = remove_prints_from_file(dart_file)
        total_removed += removed
        files_processed += 1

    print("=" * 60)
    print(f"✅ Complete! Processed {files_processed} files")
    print(f"📊 Total print statements removed: ~{total_removed}")
    print("")
    print("Run this to verify:")
    print("  grep -r 'print(' ruwaq_jawi/lib/core --include='*.dart' | wc -l")

if __name__ == "__main__":
    main()

File: remove_prints.js
#!/usr/bin/env node
/**
 * Remove all print() and debugPrint() statements from lib/core folder
 * Preserves if (kDebugMode) blocks but removes only the prints inside
 * Skips app_logger.dart as it's a logging utility
 */

const fs = require('fs');
const path = require('path');

function getAllDartFiles(dir, fileList = []) {
  const files = fs.readdirSync(dir);

  files.forEach(file => {
    const filePath = path.join(dir, file);
    const stat = fs.statSync(filePath);

    if (stat.isDirectory()) {
      getAllDartFiles(filePath, fileList);
    } else if (file.endsWith('.dart')) {
      fileList.push(filePath);
    }
  });

  return fileList;
}

function removePrintStatements(content) {
  let modified = content;
  let count = 0;

  // Pattern 1: Match simple print statements on single lines
  // print('...');  or  print("...");
  const pattern1 = /(\s*)print\([^;]*?\);/g;
  const matches1 = modified.match(pattern1) || [];
  count += matches1.length;
  modified = modified.replace(pattern1, '$1// Debug logging removed');

  // Pattern 2: Match debugPrint statements
  const pattern2 = /(\s*)debugPrint\([^;]*?\);/g;
  const matches2 = modified.match(pattern2) || [];
  count += matches2.length;
  modified = modified.replace(pattern2, '$1// Debug logging removed');

  // Pattern 3: Match multi-line print statements (more aggressive)
  // This handles cases where print spans multiple lines
  const pattern3 = /(\s*)print\([^)]*\n[^)]*\);/g;
  modified = modified.replace(pattern3, '$1// Debug logging removed');

  // Pattern 4: Match multi-line debugPrint statements
  const pattern4 = /(\s*)debugPrint\([^)]*\n[^)]*\);/g;
  modified = modified.replace(pattern4, '$1// Debug logging removed');

  return { modified, count };
}

function processFile(filePath) {
  // Skip app_logger.dart - it's a logging utility
  if (filePath.includes('app_logger.dart')) {
    console.log(`⏭️  Skipping ${filePath} (logging utility)`);
    return 0;
  }

  try {
    const content = fs.readFileSync(filePath, 'utf8');
    const { modified, count } = removePrintStatements(content);

    if (modified !== content) {
      fs.writeFileSync(filePath, modified, 'utf8');
      console.log(`✅ ${filePath}: Removed ~${count} print statements`);
      return count;
    } else {
      console.log(`⏭️  ${filePath}: No prints found`);
      return 0;
    }
  } catch (error) {
    console.error(`❌ Error processing ${filePath}:`, error.message);
    return 0;
  }
}

function main() {
  const corePath = path.join(__dirname, 'ruwaq_jawi', 'lib', 'core');

  if (!fs.existsSync(corePath)) {
    console.error(`❌ Path not found: ${corePath}`);
    process.exit(1);
  }

  console.log('🚀 Starting print statement removal from lib/core');
  console.log('='.repeat(60));

  const dartFiles = getAllDartFiles(corePath);
  let totalRemoved = 0;
  let filesProcessed = 0;

  dartFiles.forEach(file => {
    const removed = processFile(file);
    totalRemoved += removed;
    filesProcessed++;
  });

  console.log('='.repeat(60));
  console.log(`✅ Complete! Processed ${filesProcessed} files`);
  console.log(`📊 Total print statements removed: ~${totalRemoved}`);
  console.log('');
  console.log('Run this to verify:');
  console.log('  grep -r "print(" ruwaq_jawi/lib/core --include="*.dart" | wc -l');
}

main();
